project/frontend/src/components/ResultsMap.js

```diff
@@ -1,19 +1,29 @@
+// TODO: Change file name to MapContainer.js to match the component name.
 import React, { Fragment, useState } from 'react';
 import GoogleMapReact from 'google-map-react';
 import Lunch from '../assets/lunch.svg';
 
 function MapContainer(props) {
-  const restaurants = props.restaurants;
-  const userCenter = props.userLocation;
-  const [activeMarker, setActiveMarker] = useState({});
+  const {
+    restaurants,
+    userLocation,
+    currentCardIndex,
+    setCurrentCardIndex,
+  } = props;
+  const center =
+    restaurants.length === 0
+      ? userLocation
+      : restaurants[currentCardIndex].key.latLngCoords;
   const [showInfoWindows, setShowInfoWindows] = useState({
     marker0: false,
     marker1: false,
     marker2: false,
     marker3: false,
+    marker4: false,
+    marker5: false,
+    marker6: false,
```

Comment:
Is there a way to create this using a loop? If we wanted to change the # of markers or changed the delta of 3 below where we create them, this seems like it'd be prone to falling out of sync

Reply:
+1 to this! You could do a loop from 0 to numOfMarkers and then just add a key value pair to the object in each iteration

Reply:
it turns out i don't even need a for loop! The functions that I have for onMouseEnterMarker and onMouseLeaveMarker handle adding these properties! And is dynamic depending on the size of the restaurant list.